tests/integration/test.test.ts

```diff
@@ -0,0 +1,157 @@
+import parseContentSecurityPolicy from "content-security-policy-parser";
+import * as cheerio from "cheerio";
+
+import { afterAll, beforeAll, expect, it, describe } from "vitest";
+import { serve } from "./helpers";
+
+let baseURL: string;
+
+let cleanup: () => Promise<void>;
+
+beforeAll(async () => {
+  [baseURL, cleanup] = await serve({
+    waitForFunctions: ["__csp-nonce"],
+  });
+  afterAll(cleanup);
+});
```

Comment:
```suggestion
beforeAll(async () => {
  [baseURL, cleanup] = await serve({
    waitForFunctions: ["__csp-nonce"],
  });
});

afterAll(cleanup);
```

Reply:
You can probably convert these to `before` and `after` to make the tests run faster without losing anything: We're not cleaning the output directory between each test, so we're not dealing with a perfectly clean fixture environment, anyway.

Reply:
@ndhoule `vitest` does not have a `before` or an `after` export, I think they are named `beforeAll` and `afterAll` - they only run once, unline `beforeEach` and `afterEach`

Reply:
With regards to the code suggestion to move the `afterAll` out into the top-level, that won't work because it means the `cleanup` variable will be `undefined` when the `afterAll` executes - If you do try this TypeScript will also report an error:
>Variable `cleanup` is used before being assigned.

Reply:
You can use a non-null assertion (or an actual runtime assert) when you use the variable:

```suggestion
beforeAll(async () => {
  [baseURL, cleanup] = await serve({
    waitForFunctions: ["__csp-nonce"],
  });
  afterAll(cleanup!);
});
```

Alternatively, you can check the variable before use:

```suggestion
beforeAll(async () => {
  [baseURL, cleanup] = await serve({
    waitForFunctions: ["__csp-nonce"],
  });
  afterAll(async () => {
    if (cleanup !== undefined) {
      await cleanup();
    }
  });
});
```

I normally avoid non-null assertions, but in this case it's fine and probably the most idiomatic option--TypeScript can't statically understand the semantics of `beforeAll`.